DSharpPlus.SlashCommands/Attributes/MinimumLengthAttribute.cs

```diff
@@ -0,0 +1,47 @@
+﻿// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2022 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+
+namespace DSharpPlus.SlashCommands
+{
+    /// <summary>
+    /// Sets a minimum allowed length for this slash command option. Only valid for <see cref="string"/> parameters.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+    public class MinimumLengthAttribute : Attribute
+    {
+        /// <summary>
+        /// The value.
+        /// </summary>
+        public int Value { get; internal set; }
+
+        /// <summary>
+        /// Sets a minimum allowed length for this slash command option. Only valid for <see cref="string"/> parameters.
+        /// </summary>
+        public MinimumLengthAttribute(int value)
+        {
+            this.Value = value;
+        }
+    }
+}
```

Comment:
Min and Max length attributes should be the same attribute (MinMaxLengthAttribute, SpecificLengthAttribute, StringLengthAttribute, etc) for fewer reflection calls and more importantly built-in client-side verification (as we know that the min will always be required to be lesser than the max). I stress the client-side verification as it will make stack traces slightly easier to understand and will allow to easily spot errors before the bot connects to Discord

Reply:
the existing minimum and maximum attributes for integers are separated, imo consistency makes more sense here. what we could do in a separate PR is marking these as `[Obsolete]` and `[EditorBrowsable(EditorBrowsableState.Never)]`

Reply:
Sounds good to me